server/src/test/java/keywhiz/service/daos/SecretSeriesDAOTest.java

```diff
@@ -47,6 +50,51 @@
   @Inject @Readwrite SecretContentDAO secretContentDAO;
   @Inject @Readwrite GroupDAO groupDAO;
 
+  @Test public void renameUpdatesRowHmac() {
+    long id = createRandomSecretSeries();
+    String originalHmac = getRowHmac(id);
+
+    String newName = randomName();
+    secretSeriesDAO.renameSecretSeriesById(id, newName, "creator", ApiDate.now().toEpochSecond());
+    String renamedHmac = getRowHmac(id);
+
+    assertNotEquals(originalHmac, renamedHmac);
+  }
+
+  @Test public void createSecretSeriesUsesSuppliedId() {
+    long wellKnownId = 123;
+
+    long id = secretSeriesDAO.createSecretSeries(
+        wellKnownId,
+        "abc",
+        null,
+        null,
+        null,
+        null,
+        null,
+        ApiDate.now().toEpochSecond());
+
+    assertEquals(wellKnownId, id);
+  }
+
+  @Test public void computesWellKnownRowHmac() {
+    long wellKnownId = 123;
+
+    secretSeriesDAO.createSecretSeries(
+        wellKnownId,
+        "abc",
+        null,
+        null,
+        null,
+        null,
+        null,
+        ApiDate.now().toEpochSecond());
+
+    assertEquals(
+        "649F23AF8C88F32AAEDD848EB5F42A8770BBE59B6DBE20E66DDA2AB915E28876",
+        getRowHmac(wellKnownId));
+  }
```

Comment:
This known-answer test actually caught a bug where I had reversed the order of the inputs to the HMAC (secret series ID and name) during refactoring. Regardless, I wanted to add this as a warning in case future maintainers inadvertently (or otherwise) modify how the HMAC is calculated.

Reply:
Might be good to leave this cautionary tale as a comment :P